KILTprotocol/sdk-js
# docs: update to Substrate 2.0 compliance

## fixes KILTProtocol/ticket#543
- updates documentation to be compliant with current state of the SDK
- adds _getting-started.ts_ to mirror code from _getting-started.md_
- adds _getting-started.ts_ to CI for checking that _getting-started.md_ only includes working code

For review please focus on _getting-started.md_. For each code snippet in the markdown I added a comment of form `/* ... */` to the TypeScript file including the last line of the markdown file before displaying the code snippet.

#### Example

##### getting-started.md

The Attester creates the attestation based on the IRequestForAttestation object she received:

```typescript
const attestation = Kilt.Attestation.fromRequestAndPublicIdentity(
  extractedRequestForAttestation.content.requestForAttestation,
  attester.getPublicIdentity()
)
```
##### getting-started.ts

```typescript
    /* The Attester creates the attestation based on the IRequestForAttestation object she received: */
    const attestation = Kilt.Attestation.fromRequestAndPublicIdentity(
      extractedRequestForAttestation.content.requestForAttestation,
      attester.getPublicIdentity()
    )
```

## How to test:
Check that code in _getting-started.md_ matches the one in the corresponding TypeScript file. 

```bash
docker run -p 9944:9944 -d kiltprotocol/mashnet-node:develop ./target/release/mashnet-node --dev --ws-port 9944 --ws-external --rpc-external
yarn ts-node docs/getting-started.ts
docker kill $(docker ps -q)
```

## Checklist:
